nextcord/user.py

```diff
@@ -89,13 +94,16 @@ def __hash__(self) -> int:
     def _update(self, data: Union[PartialUserPayload, UserPayload]) -> None:
         self.name = data["username"]
         self.id = int(data["id"])
-        self.discriminator = data["discriminator"]
+        self._discriminator = data.get("discriminator")
         self._avatar = data["avatar"]
         self._banner = data.get("banner", None)
         self._accent_colour = data.get("accent_color", None)
         self._public_flags = data.get("public_flags", 0)
         self.bot = data.get("bot", False)
         self.system = data.get("system", False)
+        self.global_name = data.get("global_name", None)
+
+        self.discriminator = self._discriminator if self._discriminator != "0" else "0"
```

Comment:
this is just `y = "0" if x == "0" else x`  (swapped condition) which is semantically the same as just `x`

Reply:
(JFC) This was initially because `discriminator` was going to be `str | None`, but I must have forgotten to change that from https://github.com/nextcord/nextcord/pull/1061#discussion_r1237846949